src/solvers/petsc_linear_solver.C

```diff
@@ -1262,7 +1262,15 @@ void PetscLinearSolver<T>::get_residual_history(std::vector<double> & hist)
   // methods, the number of residuals returned in the history
   // vector may be different from what you are expecting.  For
   // example, TFQMR returns two residual values per iteration step.
+
+  // Recent development versions of PETSc require the residual
+  // history vector pointer to be declared as const
+#if PETSC_VERSION_RELEASE == 0
```

Comment:
Not quite right. We need to use version comparison  as well 

!PETSC_RELEASE_LESS_THAN(major,minor,subminor) should work

Reply:
Unfortunately, the major, minor and subminor version numbers are the same for the release version and the current PETSc git master. I'm sure it will change in the near future, but for now the only difference is the PETSC_VERSION_RELEASE info.

I added the PETSC_RELEASE_LESS_THAN check in my last commit. I assume the const declaration will be used in PETSc 3.14.4 and above.

Reply:
Yeah, but what version should we put? Is it guaranteed to be in 3.15.x since it's in master now and it's not in the 3.14.x series? I can take care of making this change...

Reply:
> I can take care of making this change...

Oops, I see that @lcnsqr already fixed it, but note that as @fdkong says we should use `PETSC_RELEASE_LESS_THAN(3,14,4);` which does the same logic as your check.